File: src/com/music/read/LogCat.java
package com.music.read;

/**
 * Created by xupanpan on 5/16/17.
 */
public class LogCat {


    public static final int I = 0;
    public static final int D = 1;
    public static int LOG_LEVER = 0;


    public static void i(String str, String str1, String str2) {
        i(str + " " + str1 + " " + str2);
    }

    public static void i(String str, String str1) {

        i(str + " " + str1);
    }

    public static void i(String str) {
        if (I == LOG_LEVER) {
            print("I", str);
        }
    }

    public static void d(String str, String str1, String str2) {
        d(str + " " + str1 + " " + str2);
    }

    public static void d(String str, String str1) {

        d(str + " " + str1);
    }

    public static void d(String str) {
        if (D == LOG_LEVER) {
            print("D", str);
        }
    }


    public static void print(String logLever, String str) {

        String time = TimeUtils.convertTime(System.currentTimeMillis(), TimeUtils.FORMAT_SECONDS3);

        String threadInfo = Thread.currentThread().getName() + ":" + Thread.currentThread().getId();

        System.out.println(time + " " + threadInfo + "/? " + logLever + "/ " + str);

    }

}

File: src/com/music/read/LogCat.kt
package com.music.read

/**
 * Created by xupanpan on 5/16/17.
 */
object LogCat {


    val I = 0
    val D = 1
    var LOG_LEVER = 0


    fun i(str: String, str1: String, str2: String) {
        i("$str $str1 $str2")
    }

    fun i(str: String, str1: String) {

        i(str + " " + str1)
    }

    fun i(str: String) {
        if (I == LOG_LEVER) {
            print("I", str)
        }
    }

    fun d(str: String, str1: String, str2: String) {
        d("$str $str1 $str2")
    }

    fun d(str: String, str1: String) {

        d(str + " " + str1)
    }

    fun d(str: String) {
        if (D == LOG_LEVER) {
            print("D", str)
        }
    }


    fun print(logLever: String, str: String) {

        val time = TimeUtils.convertTime(System.currentTimeMillis(), TimeUtils.FORMAT_SECONDS3)

        val threadInfo = Thread.currentThread().name + ":" + Thread.currentThread().id

        println("$time $threadInfo/? $logLever/ $str")

    }

}
